qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java

```diff
@@ -496,18 +496,41 @@ public static Boolean mandatoryFieldsAreAnswered(Map<String, Ask> map, BaseEntit
 			String value = baseEntity.getValueAsString(attributeCode);
 
 			// if any are blank, mandatory and non-readonly, then task is not complete
-			if ((mandatory && !readonly) && StringUtils.isBlank(value))
-				answered = false;
+			log.debug("Checking attrib: " + attributeCode + ": readonly: " + readonly);
+			if ((mandatory && !readonly))
+				answered = acceptableAnswer(value);
 
-			String resultLine = (mandatory ? "[M]" : "[O]") + " : " + attributeCode + " : " + value;
-			log.debug("===> " + resultLine);
+			if(!answered) {
+				log.debug("Found unanswered mandatory: " + attributeCode);
+				break;
+			}
+
+			String resultLine = (readonly ? "[R]" : "") + (mandatory ? "[M]" : "[O]") + " : " + attributeCode + " : " + value;
```

Comment:
If it is readonly, then we are not expecting any answer, so we should be ignoring them.

Reply:
Apologies, was starting to scratch my head a bit about why this one answer was being let through supposedly